components/molecules/PhoneNumberInput.vue

```diff
@@ -0,0 +1,289 @@
+
+const phoneNumberInputEl = ref<typeof PhoneNumberInput | undefined>(undefined)
+
+const isCountryAvailable = (locale) => {
+    return countriesIso.includes(locale)
+}
+
+const props = withDefaults(
+    defineProps<{
+        value?: string
+        id?: string
+        color: string
+        errorColor: string
+        defaultCountryCode: CountryCode
+        onlyCountries?: CountryCode[] | string[]
+        ignoredCountries?: CountryCode[]
+        translations?: {} | null
+        noValidatorState?: boolean
+        error: boolean
+        noExample?: boolean
+        countriesHeight?: number
+        noUseBrowserLocale?: boolean
+        fetchCountry?: boolean
+        borderRadius?: number
+        disabled?: boolean
+    }>(),
+    {
+        value: '',
+        id: 'MazPhoneNumberInput',
+        color: 'dodgerblue',
+        errorColor: 'orangered',
+        defaultCountryCode: 'GB',
+        onlyCountries: () => [],
+        ignoredCountries: () => [],
+        translations: null,
+        noValidatorState: false,
+        error: false,
+        noExample: false,
+        countriesHeight: 30,
+        noUseBrowserLocale: false,
+        fetchCountry: false,
+        borderRadius: 4,
+        disabled: false,
+    },
+)
+
+const emit = defineEmits(['update', 'input', 'phone-number-focused', 'phone-number-blur'])
+const results = ref<Results>()
+const userLocale = ref(props.defaultCountryCode)
+const lastKeyPressed = ref(0)
+const uniqueId = computed(() => `${props.id}-${Math.random().toString(36).substring(2, 9)}`)
+
+const translatedCountryName = computed(() => ({
+    ...locales,
+    ...props.translations,
+}))
+
+const codesCountries: ComputedRef<PhoneCodeCountry[]> = computed(() => countries)
+
+const countryCode: Ref<CountryCode> = computed({
+    get() {
+        return userLocale.value || results.value?.countryCode
+    },
+    set(newCountry) {
+        setLocale(newCountry)
+        phoneNumberInputEl.value?.$el.querySelector('input').focus()
+    },
+})
+
+const phoneNumber = computed({
+    get() {
+        return props.value
+    },
+    set(newPhone) {
+        emitValues({ countryCode: countryCode.value, phoneNumber: newPhone })
+    },
+})
+
+const shouldChooseCountry = computed(() => {
+    return !countryCode.value && !!phoneNumber.value
+})
+
+const isValid = computed(() => results.value?.isValid)
+
+const phoneNumberExample = computed(() => {
+    const exampleNumber = countryCode.value ? getExampleNumber(countryCode.value, examples) : null
+    return exampleNumber?.formatNational() ?? null
+})
+
+const hasEmptyPhone = computed(() => phoneNumber.value === '' || phoneNumber.value === null)
+
+const hintValue = computed(() => {
+    return props.noExample || !phoneNumberExample.value
+        ? ''
+        : hasEmptyPhone.value || isValid.value
+          ? ''
+          : `${translatedCountryName.value.example} ${phoneNumberExample.value}`
+})
```

Comment:
This is hard to read, maybe use conditionals.

Reply:
You're right, I used conditionals instead